Awesome Nextflow
=================

A curated list of Nextflow pipelines inspired by other [awesome-*](https://github.com/bayandin/awesome-awesomeness) lists.

Example pipelines
-------------------
* [cbcrg/ampa-nf](https://github.com/cbcrg/ampa-nf) - Automated prediction of protein antimicrobial regions
* [cbcrg/mta-nf](https://github.com/cbcrg/mta-nf) - A method for best alignment of evaluation trees
* [nextflow-io/rnaseq-nf](https://github.com/nextflow-io/rnaseq-nf) - A basic pipeline for quantification of RNA-seq data using Salmon.
* [nextflow-io/rnatoy](https://github.com/nextflow-io/rnatoy) - A basic RNA-seq pipeline
* [nextflow-io/elixir-workshop-21](https://github.com/nextflow-io/elixir-workshop-21) - A basic pipeline with QC and alignment 

Featured pipelines
-------------------
* [AndersenLab/cegwas2-nf](https://github.com/AndersenLab/cegwas2-nf) - GWA mapping with C. elegans.
* [alesssia/YAMP](https://github.com/alesssia/YAMP) - Yet Another Metagenomic Pipeline.
* [anvlasova/FA-nf](https://github.com/anvlasova/FA-nf) - Functional annotation pipeline for proteins from non-model organisms.
* [biocorecrg/allele_specific_RNAseq](https://github.com/biocorecrg/allele_specific_RNAseq) - Allele-specific RNA-seq pipeline
* [bactopia/bactopia](https://github.com/bactopia/bactopia) - A flexible pipeline for complete analysis of bacterial genomes.
* [biocorecrg/ExOrthist](https://github.com/biocorecrg/ExOrthist) - ExOrthist is a Nextflow based pipeline to infer exon orthology groups at all evolutionary distances.
* [biocorecrg/indrop](https://github.com/biocorecrg/indrop)- Single cell transcriptome analysis pipeline based on DropEst.
* [biocorecrg/master_of_pores](https://github.com/biocorecrg/master_of_pores) -  Nextflow pipeline for analysis of Nanopore reads.
* [biocorecrg/MOP2](https://github.com/biocorecrg/MOP2) -  DSL2 version of master of pores. Faster and better. 
* [biocorecrg/transcriptome_assembly](https://github.com/biocorecrg/transcriptome_assembly) - De novo transcriptome assembly and annotations workflow based on trinity / transdecoder. 
* [biocorecrg/vectorQC](https://github.com/biocorecrg/vectorQC) - A Nextflow pipeline for assembling and annotating vectors. 
* [brwnj/smoove-nf](https://github.com/brwnj/smoove-nf) - Smoove workflow for Structural Variant calling and Quality Control.
* [cbcrg/kallisto-nf](https://github.com/cbcrg/kallisto-nf) - Nextflow implementation of Kallisto & Sleuth RNA-Seq Tools.
* [cbcrg/piper-nf](https://github.com/cbcrg/piper-nf) - RNA mapping pipeline.
* [cbcrg/unistrap](https://github.com/cbcrg/unistrap) - Reliability measure of inferred phylogenetic trees.
* [crickbabs/BABS-aDNASeq](https://github.com/crickbabs/BABS-aDNASeq) - An ancient DNA nextflow analysis pipeline.
* [CRG-CNAG/CalliNGS-NF](https://github.com/CRG-CNAG/CalliNGS-NF) - Variant Calling Analysis with RNA-Seq data based on GATK best practices.
* [Dowell-Lab/ChIP-Flow](https://github.com/Dowell-Lab/ChIP-Flow) - Nextflow pipeline for ChIP-seq analysis.
* [Dowell-Lab/RNAseq-Flow](https://github.com/Dowell-Lab/RNAseq-Flow) - Nextflow pipeline for processing RNA-seq data.
* [Dowell-Lab/Nascent-Flow](https://github.com/Dowell-Lab/Nascent-Flow) - Nascent Transcription Processing Pipeline.
* [FredHutch/reproducible-workflows](https://github.com/FredHutch/reproducible-workflows/tree/master/nextflow) - Reproducible Workflows, curated at the Fred Hutch.
* [guigolab/chip-nf](https://github.com/guigolab/chip-nf) - Automated ChIP-seq pipeline.
* [guigolab/grape-nf](https://github.com/guigolab/grape-nf) - Automated RNA-seq pipeline.
* [guigolab/ipsa-nf](https://github.com/guigolab/ipsa-nf) - Integrative Pipeline for Splicing Analyses.
* [guigolab/FA-nf](https://github.com/guigolab/FA-nf) - Functional annotation pipeline for proteins from non-model organisms 
* [h3abionet/h3abionet16S](https://github.com/h3abionet/h3abionet16S) - Integrated pipeline for 16S rDNA diversity analysis.
* [JaneliaSciComp/nextflow-spark](https://github.com/JaneliaSciComp/nextflow-spark)
* [IARC/needlestack](https://github.com/IARCbioinfo/needlestack) - Multi-sample somatic variant caller.
* [IARC/gatk4-GenotypeGVCFs-nf](https://github.com/IARCbioinfo/gatk4-GenotypeGVCFs-nf) - Joint calling of gVCF, following GATK4 Best Practices.
* [IARC/gatk4-HaplotypeCaller-nf](https://github.com/IARCbioinfo/gatk4-HaplotypeCaller-nf) - GATK4 HaplotypeCaller step, in gVCF mode, first step for subsequent whole cohort Joint Genotyping.
* [icgc-argo/sanger-wgs-variant-calling](https://github.com/icgc-argo/sanger-wgs-variant-calling) - Sanger WGS Variant Caller.
* [jdidion/atropos](https://github.com/jdidion/atropos/tree/main/paper) - NGS read trimming tool that is specific, sensitive, and speedy.
* [labsyspharm/mcmicro](https://github.com/labsyspharm/mcmicro) - End-to-end pipeline for processing multiplexed whole slide imaging and tissue microarrays.
* [loosolab/TOuCAN](https://github.molgen.mpg.de/loosolab/TOuCAN) - Targeted chrOmatin Capture ANalysis.
* [metagenomics/MeRaGENE](https://github.com/metagenomics/MeRaGENE) - Metagenomics rapid gene identification pipeline.
* [mirnylab/distiller-nf](https://github.com/mirnylab/distiller-nf) - Modular Hi-C mapping pipeline.
* [montilab/pipeliner](https://github.com/montilab/pipeliner) - Framework for the definition of sequencing data processing workflows.
* [NCBI-Hackathons/ATACFlow](https://github.com/NCBI-Hackathons/ATACFlow) - ATAC-seq pipeline wrapped in NextFlow.
* [NCSA/GenomicsCortextVarNextflow](https://github.com/ncsa/GenomicsCortextVarNextflow) - Structural Variant Calling with Cortex-Var.
* [nf-core/chipseq](https://github.com/nf-core/chipseq) - Chromatin immunoprecipitation (ChIP-seq) peak-calling, QC and differential analysis pipeline.
* [nf-core/eager](https://github.com/nf-core/eager) - A fully reproducible and state of the art ancient DNA analysis pipeline.
* [nf-core/hlatyping](https://github.com/nf-core/hlatyping) - Precision HLA typing from next-generation sequencing data.
* [nf-core/lncpipe](https://github.com/nf-core/lncpipe) - Analysis of long non-coding RNAs from RNA-seq datasets.
* [nf-core/mag](https://github.com/nf-core/mag) - Assembly and binning of metagenomes.
* [nf-core/methylseq](https://github.com/nf-core/methylseq) - Methylation (Bisulfite-Sequencing) analysis pipeline using Bismark or bwa-meth + MethylDackel.
* [nf-core/pangenome](https://github.com/nf-core/pangenome) - Renders a collection of sequences into a pangenome graph.
* [nf-core/rnaseq](https://github.com/nf-core/rnaseq) - RNA sequencing analysis pipeline using STAR, HISAT2 and Salmon with gene counts and quality control.
* [nf-core/rnafusion](https://github.com/nf-core/rnafusion) - RNA sequencing analysis pipeline with curated list of tools for detecting and visualizing fusion genes.
* [nf-core/sarek](https://github.com/nf-core/sarek) - Analysis pipeline to detect germline or somatic variants from WGS / targeted sequencing.
* [nf-core/smrnaseq](https://github.com/nf-core/smrnaseq) - A small-RNA sequencing analysis pipeline.
* [nf-core/vipr](https://github.com/nf-core/vipr) - Assembly and intrahost / low-frequency variant calling for viral samples.
* [nf-core/viralrecon](https://github.com/nf-core/viralrecon/) - Assembly and intrahost/low-frequency variant calling for viral samples.
* [nmdp-bioinformatics/flow](https://github.com/nmdp-bioinformatics/flow) - Consensus assembly and variant calling workflow.
* [robsyme/nf-repeatmasking](https://github.com/robsyme/nf-repeatmasking) - Workflow for automatic repeat detection, classification and masking.
* [qbicsoftware/icgc-featurecounts](https://github.com/qbicsoftware/icgc-featurecounts) - Pipeline to run featureCounts on RNAseq BAM files.
* [sanger-pathogens/companion](https://github.com/sanger-pathogens/companion) - Eukaryotic genome annotation pipeline.
* [tburk/smallRNA-meth](https://gitlab.com/tburk/smallRNA-meth) - [Analyzes smallRNA methylation data](https://www.nature.com/articles/nmeth.4610).
* [ZuberLab/crispr-process-nf](https://github.com/ZuberLab/crispr-process-nf) - Process CRISPR and shRNA functional genetic screening data.
* [ZuberLab/mageck-nf](https://github.com/ZuberLab/mageck-nf) - Statistical Analysis of multiplexed CRISPR / shRNA Screens using MAGeCK.

Covid-19 related projects
-------------------------
* [biocorecrg/master_of_pores](https://github.com/biocorecrg/master_of_pores) - Nextflow pipeline for analysis of Nanopore data from direct RNA sequencing (see [also](https://biocorecrg.github.io/covid/)).
* [BU-ISCIII/SARS-Cov2_analysis](https://github.com/BU-ISCIII/SARS-Cov2_analysis) - SARS-Cov2 bioinformatics analysis.
* [connor-lab/ncov2019-artic-nf](https://github.com/connor-lab/ncov2019-artic-nf) - A Nextflow pipeline for running the ARTIC network's fieldbioinformatics tools with a focus on ncov2019.
* [heuermh/sars-cov-2](https://github.com/heuermh/sars-cov-2) - Transform SARS-CoV-2 annotated genomes into sequences and features in [Apache Parquet](http://parquet.apache.org) format.
* [jnoms/virID](https://github.com/jnoms/virID) - Viral Identification and Discovery - A viral characterization pipeline built in Nextflow.
* [nf-core/viralrecon](https://github.com/nf-core/viralrecon/) - Assembly and intrahost/low-frequency variant calling for viral samples.
* [vtilloy/aspicov](https://gitlab.com/vtilloy/aspicov) - Automated and Standardized Pipeline for Identification of SARS-Cov2 nucleotidic Variants.

Other pipelines
------------------
* [abreschi/ShortRNA-nf](https://github.com/abreschi/ShortRNA-nf) - Pipeline for processing shortRNA-seq data.
* [andremrsantos/next-pipes-lghm](https://github.com/andremrsantos/next-pipes-lghm) - Eukaryotic genome annotation pipeline.
* [AveraSD/nextflow-kallisto](https://github.com/AveraSD/nextflow-kallisto)
* [AveraSD/nextflow-rnastar](https://github.com/AveraSD/nextflow-rnastar)
* [cbcrg/benchfam](https://github.com/cbcrg/benchfam)
* [cbcrg/grape-nf](https://github.com/cbcrg/grape-nf) - Yet another RNA-Seq pipeline (deprecated)
* [cerebis/meta-sweeper](https://github.com/cerebis/meta-sweeper) - Parametric sweep of simulated microbial communities and metagenomic sequencing.
* [charite/ngspipeline](https://github.com/charite/ngspipeline)
* [chrisquince/DESMAN](https://github.com/chrisquince/DESMAN) - De novo Extraction of Strains from MetAgeNomes.
* [evanfloden/concTree](https://github.com/evanfloden/concTree) - Creates consensus trees using alignment uncertainty from many MSAs.
* [evanfloden/lncRNA-Annotation-nf](https://github.com/evanfloden/lncRNA-Annotation-nf) - lncRNA Annotation Pipeline based on STAR, Cufflinks and FEELnc.
* [evanfloden/paraMSA](https://github.com/evanfloden/paraMSA/) - Multiple sequence alignments for phylogenetic trees and bootstrap support values.
* [evanfloden/tuxedo-nf](https://github.com/evanfloden/tuxedo-nf) - Nextflow implementation of the Tuxedo Suite of Tools.
* [evanfloden/vast-tools-nf](https://github.com/evanfloden/vast-tools-nf) - Nextflow Implementation of VAST-TOOLS for profiling alternative splicing events in RNA-Seq data.
* [fredericlemoine/rna-pipeline](https://github.com/fredericlemoine/rna-pipeline)
* [h3abionet/chipimputation](https://github.com/h3abionet/chipimputation) - Imputation workflow developed as part of the H3 Africa BioNet Hackathon.
* [h3abionet/h3agwas](https://github.com/h3abionet/h3agwas) - GWAS pipeline developed as part of the H3 Africa BioNet Hackathon.
* [HadrienG/pipelines](https://github.com/HadrienG/pipelines)
* [hmkim/workflow](https://github.com/hmkim/workflow/tree/master/nextflow)
* [holtgrewe/ngs_pipelines](https://github.com/holtgrewe/ngs_pipelines) - Nextflow-based pipelines for typical NGS processing tasks.
* [IARCbioinfo/bam_realignment-nf](https://github.com/IARCbioinfo/bam_realignment-nf)
* [IARCbioinfo/bametrics-nf](https://github.com/IARCbioinfo/bametrics-nf) - Compute average metrics from reads that overlap a given set of positions.
* [IARCbioinfo/GVCF_pipeline-nf](https://github.com/IARCbioinfo/GVCF_pipeline-nf)
* [IARCbioinfo/mpileup-nf](https://github.com/IARCbioinfo/mpileup-nf) - Coverage computation with samtools mpileup.
* [IARCbioinfo/mutect-nf](https://github.com/IARCbioinfo/mutect-nf)
* [InSilicoDB/pipeline-kallisto](https://github.com/InSilicoDB/pipeline-kallisto/)
* [InSilicoDB/snp-imputation-nf](https://github.com/InSilicoDB/snp-imputation-nf)
* [jdidion/rna-quick](https://github.com/jdidion/rna-quick) - Rapid analysis of gene and transcript quantification and differential expression.
* [joshua-d-campbell/nf-GATK_Exome_Preprocess](https://github.com/joshua-d-campbell/nf-GATK_Exome_Preprocess) - GATK best practice guide to preprocess whole exome sequencing (WES) data.
* [meissnert/nextflow-gtool](https://github.com/meissnert/nextflow-gtool)
* [CDCgov/mycosnp-nf](https://github.com/CDCgov/mycosnp-nf) - Portable workflow for performing whole genome sequencing analysis of fungal organisms.
* [NationalGenomicsInfrastructure/icing](https://github.com/NationalGenomicsInfrastructure/icing) - HLA typing from OxfordNanopore reads.
* [NBISweden/wgs-structvar](https://github.com/NBISweden/wgs-structvar) - Genome Sequenceing Structural Variation Pipelines.
* [MHH-RCUG/nf_wochenende](https://github.com/MHH-RCUG/nf_wochenende) - A metagenomic (and genomic) alignment and normalization pipeline for long and short reads
* [nmdp-bioinformatics/flow-blast-hml](https://github.com/nmdp-bioinformatics/flow-blast-hml)
* [nmdp-bioinformatics/flow-validation](https://github.com/nmdp-bioinformatics/flow-validation)
* [ODiogoSilva/innuca-nf](https://github.com/ODiogoSilva/innuca-nf) - A nextflow implementation of INNUENDO quality control of reads, de novo assembly and contigs quality assessment.
* [pbelmann/sra-download](https://github.com/pbelmann/sra-download)
* [CDCgov/PHoeNIx](https://github.com/CDCgov/phoenix) - A short-read pipeline for healthcare-associated and antimicrobial resistant pathogens.
* [PlantandFoodResearch/VariantAnalysis](https://github.com/PlantandFoodResearch/VariantAnalysis) - Bioinformatic variant calling pipeline.
* [robsyme/nextflow-annotate](https://github.com/robsyme/nextflow-annotate) - Fungal genome annotation workflow
* [robsyme/nextflow-mapping](https://github.com/robsyme/nextflow-mapping)
* [robsyme/workflows-proteogenomics](https://github.com/robsyme/workflows-proteogenomics) - Versioned, shareable and replicable workflows for proteogenomics.
* [CDCgov/SC2CLIA](https://github.com/CDCgov/SC2CLIA) - Pipeline for Clinical Laboratories Improvements Amendments (CLIA) compliant variant calling and spike protein substitution prediction with additional tools for quality control
* [skner/iasi-pipe](https://github.com/skner/iasi-pipe)
* [snewhouse/ngs_nextflow](https://github.com/snewhouse/ngs_nextflow) - Alignment, Variant Calling, GATK Best Practices.
* [wittelab/orchid](https://github.com/Wittelab/orchid) - Novel management, annotation, and machine learning framework for analyzing cancer mutations.

Tutorials
----------
* [Nextflow training portal by Seqera](https://training.seqera.io)
* [Bioinformatics pipeline example from the bottom up](https://ricomnl.com/blog/bottom-up-bioinformatics-pipeline/)
* [CRG course](https://github.com/nextflow-io/crg-course-nov16)
* [Introduction to Bioinformatics workflows with Nextflow and nf-core](https://carpentries-incubator.github.io/workflows-nextflow/)
* [Nextflow patterns](https://github.com/nextflow-io/patterns) - A collection of common Nextflow implementation patterns
* [Nextflow tutorial](https://github.com/nextflow-io/hack17-tutorial)
* [Nextflow demos](https://github.com/stevekm/nextflow-demos) - Example Nextflow pipelines and programming techniques
 (by Stephen Kelly)
* [H3ABioNet course](https://github.com/shaze/nextflow-course)
* [Running nf-core pipelines with AWS Batch](https://apeltzer.github.io/post/01-aws-nfcore/)
* [Running nextflow on AWS Batch](https://antunderwood.gitlab.io/bioinformant-blog/posts/running_nextflow_on_aws_batch/)
* [How to make a bigger custom AMI?](https://maxulysse.github.io/2018/11/05/Custom-AMI/)
* [Nextflow on AWS Batch](https://aws-samples.github.io/aws-genomics-workflows/orchestration/nextflow/nextflow-overview.html)
* [Pipelines on AWS](https://t-neumann.github.io/pipelines/AWS-pipeline/)
* [Reproducible, scalable, and shareable analysis workflows with Nextflow](https://sateeshperi.github.io/nextflow_varcal/nextflow/)
* [Running Nextflow on Google Cloud](https://cloud.google.com/life-sciences/docs/tutorials/nextflow)
* [Using AWS Batch to process 67,000 genomes with Bactopia](https://read-lab-confederation.github.io/blog/posts/bactopia-aws-and-67000-genomes/)
* [Nextflow Beginners Workshop](https://github.com/vagkaratzas/nextflow-beginners-workshop)
* [nf-core Pre-Hackathon Training 2025](https://github.com/Zemzemfiras1/nf-core-pre-hackathon_training2025)
* [nextflow-training - by Aubin THOMAS IGH Montpellier](https://gitlab.in2p3.fr/ighbioinfo/nextflow-training)

Pipelines collections
-------------
* [IARC-nf](https://github.com/IARCbioinfo/IARC-nf) - List of [IARC](https://www.iarc.fr/) bioinformatics nextflow pipelines.
* [nf-core](http://nf-co.re/pipelines) - A collection of high quality Nextflow pipelines.

Modules collections
--------------------
* [biocorecrg/BioNextflow](https://github.com/biocorecrg/BioNextflow) - Repository to host DSL2 based tool-specific sub-workflows 
* [nf-core/modules](https://github.com/nf-core/modules/) - Repository to host tool-specific module files for the Nextflow DSL2 community
* [UMCUGenetics/NextflowModules](https://github.com/UMCUGenetics/NextflowModules) - UMCU Genetics Nextflow modules

Presentations
-------------
* [rsuchecki.github.io](https://rsuchecki.github.io/reproducible/show.html#/) - Reproducible and Reusable publication and analysis workflow by [bioinforad](https://twitter.com/bioinforad)
* [CIC Show-and-Tell](https://cdn.rawgit.com/robsyme/nextflow-intro/4615e5a/index.html) - Nice Nextflow presentation by [@robsyme](https://github.com/robsyme)

Video
-----
* [Hello Nextflow training series](https://www.youtube.com/playlist?list=PLPZ8WHdZGxmXiHf8B26oB_fTfoKQdhlik)
* [Nextflow: Scalable, Shareable & Reproducible workflows](https://www.youtube.com/watch?v=lqm-VV5dOgk) - Awesome intro by Rad Suchecki
* [Nextflow Youtube channel](https://www.youtube.com/channel/UCB-5LCKLdTKVn2F4V4KlPbQ)
* [Using NextFlow pipelines at Babraham](https://www.youtube.com/watch?v=oAjZ0xmewp8)

Syntax highlighting
-------------------
* [Atom](https://atom.io/packages/language-nextflow)
* [Emacs](https://github.com/Emiller88/nextflow-mode)
* [Sublime Text](https://github.com/peterk87/sublime-nextflow)
* [VIM](https://github.com/LukeGoodsell/nextflow-vim)
* [VSCode](https://github.com/nextflow-io/vscode-language-nextflow)

Other links  
---------------
* [Nextflow Hackathon '17](https://github.com/nextflow-io/hack17) - Projects repositorty of the Nextflow Hackathon held in Barcelona, Sept 2017
* [Nextflow Workbench](http://campagnelab.org/software/nextflow-workbench/) - Workbench that make it easier to write data analysis workflows with Nextflow
* [nf-core/tools](https://github.com/nf-core/tools) - Helper tools for the nf-core community, linting, template...
* [Make to Nextflow](https://github.com/lindenb/xml-patch-make/wiki/Tabix) - Converts Makefile to Nextflow
* [NCBI-Hackathons/SPeW](https://github.com/NCBI-Hackathons/SPeW) - A discussion comparting different workflow frameworks including Nextflow.
* [NGS workflow](https://jmazz.me/blog/NGS-Workflows) Excellent blog post comparing Make/Snakemake/Nextflow by Julian Mazzitelli
